components/org.wso2.carbon.identity.oauth/src/test/java/org/wso2/carbon/identity/oauth2/dao/AuthorizationCodeDAOImplTest.java

```diff
@@ -0,0 +1,446 @@
+        List<AuthzCodeDO> revokeAuthzCodeList = new ArrayList<>();
+        revokeAuthzCodeList.add(authzCodeDO1);
+        revokeAuthzCodeList.add(authzCodeDO2);
+        authorizationCodeDAO.deactivateAuthorizationCodes(revokeAuthzCodeList);
+
+        Assert.assertTrue(authorizationCodeDAO.getActiveAuthorizationCodesByConsumerKey
+                (authzCodeDO1.getConsumerKey()).isEmpty());
+        Assert.assertTrue(authorizationCodeDAO.getActiveAuthorizationCodesByConsumerKey
+                (authzCodeDO2.getConsumerKey()).isEmpty());
+    }
+
+    @Test
+    public void testGetLatestAuthorizationCodesByTenant() throws Exception {
+
+        String consumerKey1 = UUID.randomUUID().toString();
+        String authzCodeID1 = UUID.randomUUID().toString();
+        String authzCode1 = UUID.randomUUID().toString();
+        String consumerKey2 = UUID.randomUUID().toString();
+        String authzCodeID2 = UUID.randomUUID().toString();
+        String authzCode2 = UUID.randomUUID().toString();
+        persistAuthorizationCode(consumerKey1, authzCodeID1, authzCode1, true,
+                OAuthConstants.AuthorizationCodeState.ACTIVE);
+        // Creating an application outside default tenant.
+        createApplication(consumerKey2, UUID.randomUUID().toString(), 100);
+        mockStatic(OAuth2Util.class);
+        mockStatic(IdentityUtil.class);
+        when(OAuth2Util.createAuthenticatedUser(anyString(), anyString(), anyString(), anyString())).
+                thenReturn(authenticatedUser);
+
+        Assert.assertFalse(authorizationCodeDAO.getLatestAuthorizationCodesByTenant(DEFAULT_TENANT_ID).isEmpty());
+        Assert.assertTrue(authorizationCodeDAO.getLatestAuthorizationCodesByTenant(100).isEmpty());
+    }
+
+    protected void createApplication(String consumerKey, String consumerSecret, int tenantId) throws Exception {
+
+        try (PreparedStatement prepStmt = connection.prepareStatement(SQLQueries.OAuthAppDAOSQLQueries.ADD_OAUTH_APP)) {
+            prepStmt.setString(1, consumerKey);
+            prepStmt.setString(2, consumerSecret);
+            prepStmt.setString(3, USER_NAME);
+            prepStmt.setInt(4, tenantId);
+            prepStmt.setString(5, UserCoreConstants.PRIMARY_DEFAULT_DOMAIN_NAME);
+            prepStmt.setString(6, APP_NAME);
+            prepStmt.setString(7, VERSION_2);
+            prepStmt.setString(8, CALLBACK);
+            prepStmt.setString(9, "refresh_token urn:ietf:params:oauth:grant-type:saml2-bearer implicit password " +
+                    "client_credentials iwa:ntlm authorization_code urn:ietf:params:oauth:grant-type:jwt-bearer");
+            prepStmt.setLong(10, 3600L);
+            prepStmt.setLong(11, 3600L);
+            prepStmt.setLong(12, 84600L);
+            prepStmt.setLong(13, 3600L);
+            prepStmt.execute();
+            connection.commit();
+        }
+    }
+
+    protected void storeIDP() throws Exception {
```

Comment:
can't we make the access modifier private here?

Reply:
Addressed with bd9a919ce82d35f572e275a65b173e7fdf9e50dc